code/controlconfig/controlsconfig.cpp

```diff
@@ -2341,56 +2524,137 @@ int check_control(int id, int key)
+		break;
 
-	*p = 0;
-	if (Axis_map_to[1] >= 0) {
-		*p = joy_get_scaled_reading(axes_values[Axis_map_to[1]]);
+	default:
+		// ignore all others
+		break;
 	}
+}
 
-	*b = 0;
-	if (Axis_map_to[2] >= 0) {
-		*b = joy_get_scaled_reading(axes_values[Axis_map_to[2]]);
-	}
+/*!
+ * Scales, and maybe inverts, the input axis values
+ *
+ * @param[in]   bind        The control's binding to check
+ * @param[in]   action      index into axis_out of the action
+ * @param[in]   type        CC_type of the control
+ * @param[in]   frame_time  Current frame time, used to scale mouse axis
+ * @param[in]   axis_in[][] Array of raw axis values
+ * @param[out]  axis_out    Output array of the scaled axes
+ *
+ * @note C++ doesn't like passing multi-dim arrays as arguments
+ */
+void scale_invert(const CC_bind &bind,
+				int action,
+				CC_type type,
+				float frame_time,
+				int (&axis_in)[CID_JOY_MAX + 1][JOY_NUM_AXES],
+				int *axis_out)
+{
+	const int MOUSE_ID = CID_JOY_MAX;	// Joy axes go in front here, mouse gets tacked on the end
+	float factor = 0.0f;
+	int dx = 0;
+
+	switch (bind.cid) {
+	case CID_MOUSE:
+		factor = (float)Mouse_sensitivity + 1.77f;
+		factor = factor * factor / frame_time / 0.6f;
+		if (!Use_mouse_to_fly) {
+			return;
+		}
+
+		dx = axis_in[MOUSE_ID][bind.btn];
+		maybe_invert(bind.is_inverted(), type, dx);
+		axis_out[action] += (int)((float)dx * factor);
+		break;
+
+	case CID_JOY0:
+	case CID_JOY1:
+	case CID_JOY2:
+	case CID_JOY3:
+		switch (type) {
+		case CC_TYPE_AXIS_ABS:
+			dx = joy_get_unscaled_reading(axis_in[bind.cid][bind.btn]);
+			break;
 
-	*ta = 0;
-	if (Axis_map_to[3] >= 0) {
-		*ta = joy_get_unscaled_reading(axes_values[Axis_map_to[3]]);
-	}
+		case CC_TYPE_AXIS_REL:
+		case CC_TYPE_AXIS_BTN_NEG:
+		case CC_TYPE_AXIS_BTN_POS:
+		default:
+			dx = joy_get_scaled_reading(axis_in[bind.cid][bind.btn]);
+			break;
+		}
+		
+		maybe_invert(bind.is_inverted(), type, dx);
+		axis_out[action] += dx;
+		break;
 
-	*tr = 0;
-	if (Axis_map_to[4] >= 0) {
-		*tr = joy_get_scaled_reading(axes_values[Axis_map_to[4]]);
+	default:
+		// All others, ignore
+		break;
 	}
+}
 
-	if (Invert_axis[0]) {
-		*h = -(*h);
-	}
-	if (Invert_axis[1]) {
-		*p = -(*p);
-	}
-	if (Invert_axis[2]) {
-		*b = -(*b);
+void control_get_axes_readings(int *axis_v, float frame_time)
+{
+	int axe[CID_JOY_MAX + 1][JOY_NUM_AXES] = {0};
+	const int MOUSE_ID = CID_JOY_MAX;	// Joy axes go in front here, mouse gets tacked on the end
+
+	Assert(axis_v != nullptr);
+
+	// Init output
+	for (int action = 0; action < Action::NUM_VALUES; ++action) {
+		axis_v[action] = 0;
 	}
-	if (Invert_axis[3]) {
-		*ta = F1_0 - *ta;
+
+	// Read raw sticks.
+	for (short j = CID_JOY0; j < CID_JOY_MAX; ++j) {
+		joystick_read_raw_axis(j, JOY_NUM_AXES, axe[j]);
 	}
-	if (Invert_axis[4]) {
-		*tr = -(*tr);
+
+	// Read raw mouse, stuff in axes_values[0]
+	if (Use_mouse_to_fly) {
+		mouse_get_delta(&axe[MOUSE_ID][JOY_X_AXIS], &axe[MOUSE_ID][JOY_Y_AXIS], &axe[MOUSE_ID][JOY_Z_AXIS]);
```

Comment:
It's mouse code, so it should be using the mouse axes (even if they are the same values).
```suggestion
		mouse_get_delta(&axe[MOUSE_ID][MOUSE_X_AXIS], &axe[MOUSE_ID][MOUSE_Y_AXIS], &axe[MOUSE_ID][MOUSE_Z_AXIS]);
```

Reply:
At one point I made the mouse axes enums, but then later decided it may be better to have a generic set that was just "X_AXIS, Y_AXIS" etc.  I think I left it as the existing JOY_X_AXIS due to time constraints

Reply:
Ah, found the aliases. looks like I just didn't put them to use.